applications/ConvectionDiffusionApplication/python_scripts/convection_diffusion_base_solver.py

```diff
@@ -565,6 +572,9 @@ def _create_linear_solver(self):
         return linear_solver
 
     def _create_builder_and_solver(self):
+        if self.settings["analysis_type"].GetString() == "explicit":
```

Comment:
This is wrong! Analysis type must remain as `linear` or `nonlinear`. Note that you might solve a nonlinear problem using an implicit or explicit strategy (actually you are doing so).

I'd create an `time_integration` or something of the sort.

As this needs to be further discussed since it directly affects the GUI, It might be convenient to separate the changes in the Python solver in another PR to ease changing the GUI accordingly.

Reply:
If I separate this, I cannot use my tests since I am only using the "new feature" `time_integration`. Would you proceed like this?

Reply:
We can separate the Python tests from this PR and only add the cpp ones.

Reply:
There are no cpp tests. I cannot elaborate well on the reasons, since a bit of time passed after I tried to add them. However, the reason was related to this [discussion](https://github.com/KratosMultiphysics/Kratos/pull/7272#discussion_r492124259). Maybe @roigcarlo can elaborate a bit more than me, since after I did not succeed, we tried together.

Reply:
But why you can't add a cpp test checking the RHS contribution? This has nothing to do with the strategy.

Reply:
I told him to remove the tests because:

1 - tests were the only reason to make those methods public, when they should remain private.
2 - Down-casting was needed, and was ugly, don't remember exactly the lines but it could not be done directly with the derived element.

The methods are tested in python anyway, so I assumed it was enough. If its not the case, we can re-add them I suppose.